gwt/pom.xml

```diff
@@ -40,39 +40,10 @@
       <classifier>sources</classifier>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>javax.inject</groupId>
+      <artifactId>javax.inject</artifactId>
+      <classifier>sources</classifier>
```

Comment:
It just pulls the javax.inject sources JAR into the classpath (transitively, when you depend on dagger-gwt).  
This is in addition to the javax.inject "normal" JAR (transitively added through dagger "core").

In the end, a GWT app depending on dagger-gwt will have `dagger-gwt-2.x.jar`, `dagger-2.x.jar`, `dagger-2.x-sources.jar`, `javax.inject-1.jar` and `javax.inject-1-sources.jar` in its classpath (not necessarily in that order). GWT will find `dagger/Dagger.gwt.xml` and `javax/inject/Inject.gwt.xml` in the first one, `dagger/Component.class` et al. in the second one, their source files (`dagger/Component.java` et al.) in the third, `javax/inject/Provider.class` et al. in the next JAR, and their sources (`javax/inject/Provider.java` et al.) in the last JAR.

With the current (`master`) layout, the `dagger-gwt-2.x.jar` contains the `dagger/Dagger.gwt.xml`, and the JSR303 sources are bundled into the JAR into `dagger/super/`. Because of the `<super-source path="super"/>` in the .gwt.xml file, GWT then knows that it should also look for sources into the `dagger/super` "package", and will find `dagger/super/javax/inject/Provider.java` there when it looks for the sources of the `javax.inject.Provider` class (this is how one can provide different source files for GWT and for the JVM/JavaC)